### PR TITLE
Refactor Transaction NEG syncer to use strong type instead of string

### PR DESCRIPTION
**Pull request by freehan**
- Introduce a NetworkEndpoint Type to represent the meta information contained in NE.
- Introduce NetworkEndpointSet Type to support fast Get, Insert and DIfference calculation
- Adjust Transaction table to use NetworkEndpoint Type as key
- Adjust Transaction Syncer to use NetworkEndpoint and NetworkEndpointSet Type for internal plumbing. 
- Completely remove use of sets.String type for endpoint calculation.

### CODE REVIEW

File: pkg/neg/types/network_endpoint.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+func (s NetworkEndpointSet) List() []NetworkEndpoint {
+	res := make([]NetworkEndpoint, 0, len(s))
+	for key := range s {
+		res = append(res, key)
+	}
+	return []NetworkEndpoint(res)
+}
+
+// UnsortedList returns the slice with contents in random order.
+func (s NetworkEndpointSet) UnsortedList() []NetworkEndpoint {
```
Comment (MrHohn) [inline pkg/neg/types/network_endpoint.go]: remove

File: pkg/neg/syncers/transaction_table_test.go [hunk 2]
```diff
@@ -90,3 +94,11 @@ func verifyTable(t *testing.T, table transactionTable, expectTransactionMap map[
 		}
 	}
 }
+
+func genNetworkEndpoint(key string) negtypes.NetworkEndpoint {
```
Comment (MrHohn) [inline pkg/neg/syncers/transaction_table_test.go]: remove

Comment (freehan) [reply pkg/neg/syncers/transaction_table_test.go]: used in unit tests

File: pkg/neg/syncers/utils.go [hunk 3]
```diff
@@ -86,6 +86,26 @@ func calculateDifference(targetMap, currentMap map[string]sets.String) (map[stri
 	return addSet, removeSet
 }
 
+// calculateDifference determines what endpoints needs to be added and removed in order to move current state to target state.
```
Comment (MrHohn) [inline pkg/neg/syncers/utils.go]: nit: calculateDifference -> calculateNetworkEndpointDifference

**Comment by freehan** [comment]
Yes. Batch syncer will be removed in the future

**Comment by freehan** [comment]
Fixed the comments.

**Pull request merged**